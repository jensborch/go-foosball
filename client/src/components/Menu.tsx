import {
  AppBar,
  Box,
  IconButton,
  Link,
  Toolbar,
  Typography,
} from '@mui/material';
import HomeIcon from '@mui/icons-material/Home';

type MenuProps = {
  title: string;
  children?: JSX.Element;
};

const Menu = ({ title, children }: MenuProps) => {
  return (
    <AppBar position="static">
      <Toolbar>
      <IconButton
          href="/"
          component={Link}
          sx={{
            marginLeft: (theme) => theme.spacing(-1),
            marginRight: (theme) => theme.spacing(2),
          }}
          color="inherit"
          aria-label="Home"
        >
          <HomeIcon />
        </IconButton>
        <Typography variant="h6" color="inherit" sx={{ display: 'flex' }}>
          {title}
        </Typography>
        <Box
          sx={{
            position: 'absolute',
            display: 'flex',
            top: (theme) => theme.spacing(3.5),
            right: (theme) => theme.spacing(5),
          }}
        >
          {children}
        </Box>
      </Toolbar>
    </AppBar>
  );
};

export default Menu;
