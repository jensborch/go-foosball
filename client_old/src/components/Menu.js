import React from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import IconButton from '@material-ui/core/IconButton';
import HomeIcon from '@material-ui/icons/Home';
import Typography from '@material-ui/core/Typography';
import Toolbar from '@material-ui/core/Toolbar';
import { withStyles } from '@material-ui/core/styles';
import withRoot from '../withRoot';
import { Link } from 'react-router-dom';

const styles = (theme) => ({
  flex: {
    flex: 1,
  },
  menuButton: {
    marginLeft: theme.spacing.unit * -1,
    marginRight: theme.spacing.unit * 2,
  },
  buttons: {
    position: 'absolute',
    display: 'flex',
    top: theme.spacing.unit * 3.5,
    right: theme.spacing.unit * 5,
  },
});

class Menu extends React.Component {
  render() {
    const { classes } = this.props;
    return (
      <AppBar position="static">
        <Toolbar>
          <IconButton
            to="/"
            component={(props) => <Link {...props} />}
            className={classes.menuButton}
            color="inherit"
            aria-label="Home"
          >
            <HomeIcon />
          </IconButton>
          <Typography variant="h6" color="inherit" className={classes.flex}>
            {this.props.title}
          </Typography>
          <div className={classes.buttons}>{this.props.children}</div>
        </Toolbar>
      </AppBar>
    );
  }
}

Menu.propTypes = {
  classes: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
  fetch: PropTypes.func,
};

export default withRoot(withStyles(styles)(Menu));
