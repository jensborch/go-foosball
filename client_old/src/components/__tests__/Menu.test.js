import React from 'react';
import Menu from '../Menu';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';

test('Should render with title', () => {
  const component = renderer.create(
    <MemoryRouter>
      <Menu title="test" />
    </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
