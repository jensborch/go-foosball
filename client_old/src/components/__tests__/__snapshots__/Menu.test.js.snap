// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render with title 1`] = `
<header
  className="MuiPaper-root-13 MuiPaper-elevation4-19 MuiAppBar-root-4 MuiAppBar-positionStatic-8 MuiAppBar-colorPrimary-11"
>
  <div
    className="MuiToolbar-root-40 MuiToolbar-regular-42 MuiToolbar-gutters-41"
  >
    <a
      aria-label="Home"
      className="MuiButtonBase-root-50 MuiIconButton-root-44 MuiIconButton-colorInherit-45 Menu-menuButton-2"
      href="/"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiIconButton-label-49"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-53"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-98"
      />
    </a>
    <h6
      className="MuiTypography-root-62 MuiTypography-h6-79 MuiTypography-colorInherit-91 Menu-flex-1"
    >
      test
    </h6>
    <div
      className="Menu-buttons-3"
    />
  </div>
</header>
`;
